Conversion/MultiCopy.FCMacro

```diff
@@ -0,0 +1,1439 @@
+            )
+            naming_func = self._OrdinaryNumerals
+            if self._d(4, "mc_d_radiobutton_2").isChecked():
+                from_index += 1
+                to_index += 1
+                naming_func = self._UpperCaseRomanNumerals
+            elif self._d(4, "mc_d_radiobutton_3").isChecked():
+                from_index += 1
+                to_index += 1
+                naming_func = self._LowerCaseRomanNumerals
+            elif self._d(4, "mc_d_radiobutton_4").isChecked():
+                from_index += 1
+                to_index += 1
+                naming_func = self._UpperCaseAlphabet
+            elif self._d(4, "mc_d_radiobutton_5").isChecked():
+                from_index += 1
+                to_index += 1
+                naming_func = self._LowerCaseAlphabet
+            if self.dialog.findChild(QTabWidget, "mc_d_tabset").currentIndex() == 0:
+                tempBool = True
+                for selected_obj in self.selected_objs:
+                    if tempBool:
+                        for i_ in range(from_index, to_index + 1):
+                            tempStr = (
+                                selected_obj.Label
+                                + sep_text
+                                + naming_func(i_, i_ + 1)[0].zfill(pad_digits)
+                            )
+                            if len(doc2.getObjectsByLabel(tempStr)) > 0:
+                                tempBool = False
+                                break
+                    else:
+                        break
+                if not tempBool:
+                    alertMessage = "An object containing the following label already exists in the document :\n\n"
+                    alertMessage += tempStr
+                    setAlertBox(alertMessage, True)
+                    self._stop_loader()
+                    self._enable()
+                    return
+                for selected_obj in self.selected_objs:
+                    for j_ in range(from_index, to_index + 1):
+                        copy_op(selected_obj, dependency)
+                        self._rename(
+                            selected_obj,
+                            selected_obj.Label
+                            + sep_text
+                            + naming_func(j_, j_ + 1)[0].zfill(pad_digits),
+                        )
+            else:
+                execStr, execBool = self._solve_paste_code()
+                if execBool:
+                    exec(execStr)
+                else:
+                    self._stop_loader()
+                    self._enable()
+                    return
+            if del_sel:
+                for selected_obj in self.selected_objs:
+                    doc.getObject(selected_obj.Name).removeObjectsFromDocument()
+                    doc.removeObject(selected_obj.Name)
+            app.getDocument(doc2.Name).recompute()
+            app.getDocument(doc.Name).recompute()
```

Comment:
`doc` and `doc2` can be same or different; one is the `copy_doc`, and the other is the `paste_doc`.

Reply:
I didn't study the code line by line but line `copy_doc = paste_doc = app.activeDocument()`, near l.158 indicates that both variables point to the same memory address (`id(copy_doc) == id(paste_doc)`, in other words `copy_doc is paste_doc`). If `paste_doc` is then changed to point to another document, it would be good to explain this at line 158.

Reply:
I'll submit my new code. In it, `paste_doc` is sent as an argument. So, if `id(copy_doc) == id(paste_doc)`, it is because the user has chosen for it to be so; if the user selects another document to paste into, then it will be `id(copy_doc) != id(paste_doc)`. So, the variable equality only happens conditionally.